projects/groprotocol/helpers.js

```diff
@@ -0,0 +1,236 @@
+  stateMutability: "view",
+  type: "function",
+};
+const lpSuppliesAbi = {
+  constant: true,
+  inputs: [],
+  name: "totalSupply",
+  outputs: [{ internalType: "uint256", name: "", type: "uint256" }],
+  payable: false,
+  stateMutability: "view",
+  type: "function",
+};
+const token0Abi = {
+  constant: true,
+  inputs: [],
+  name: "token0",
+  outputs: [{ internalType: "address", name: "", type: "address" }],
+  payable: false,
+  stateMutability: "view",
+  type: "function",
+};
+const token1Abi = {
+  constant: true,
+  inputs: [],
+  name: "token1",
+  outputs: [{ internalType: "address", name: "", type: "address" }],
+  payable: false,
+  stateMutability: "view",
+  type: "function",
+};
+
+/* lpPositions:{
+    balance,
+    token
+}[]
+*/
+async function unwrapUniswapLPs(
+  balances,
+  lpPositions,
+  block,
+  chain = "ethereum",
+  transformAddress = (addr) => addr,
+  excludeTokensRaw = [],
+  retry = false
+) {
+  const excludeTokens = excludeTokensRaw.map((addr) => addr.toLowerCase());
+  const lpTokenCalls = lpPositions.map((lpPosition) => ({
+    target: lpPosition.token,
+  }));
+  const lpReserves = sdk.api.abi.multiCall({
+    block,
+    abi: lpReservesAbi,
+    calls: lpTokenCalls,
+    chain,
+  });
+  const lpSupplies = sdk.api.abi.multiCall({
+    block,
+    abi: lpSuppliesAbi,
+    calls: lpTokenCalls,
+    chain,
+  });
+  const tokens0 = sdk.api.abi.multiCall({
+    block,
+    abi: token0Abi,
+    calls: lpTokenCalls,
+    chain,
+  });
+  const tokens1 = sdk.api.abi.multiCall({
+    block,
+    abi: token1Abi,
+    calls: lpTokenCalls,
+    chain,
+  });
+  if (retry) {
+    await Promise.all(
+      [
+        [lpReserves, lpReservesAbi],
+        [lpSupplies, lpSuppliesAbi],
+        [tokens0, token0Abi],
+        [tokens1, token1Abi],
+      ].map(async (call) => {
+        await requery(await call[0], chain, block, call[1]);
+      })
+    );
+  }
+  await Promise.all(
+    lpPositions.map(async (lpPosition) => {
+      try {
+        const lpToken = lpPosition.token;
+        const token0 = (await tokens0).output
+          .find((call) => call.input.target === lpToken)
+          .output.toLowerCase();
+        const token1 = (await tokens1).output
+          .find((call) => call.input.target === lpToken)
+          .output.toLowerCase();
+        const supply = (await lpSupplies).output.find(
+          (call) => call.input.target === lpToken
+        ).output;
+        const { _reserve0, _reserve1 } = (await lpReserves).output.find(
+          (call) => call.input.target === lpToken
+        ).output;
+        if (!excludeTokens.includes(token0)) {
+          const token0Balance = BigNumber(lpPosition.balance)
+            .times(BigNumber(_reserve0))
+            .div(BigNumber(supply));
+          sdk.util.sumSingleBalance(
+            balances,
+            await transformAddress(token0),
+            token0Balance.toFixed(0)
+          );
+        }
+        if (!excludeTokens.includes(token1)) {
```

Comment:
`excludeTokens` logic differs from original and not compatible between them

Reply:
actually, maybe we should check `unwrapUniswapLPs` exclusion logic, seems to be broken @0xngmi @flabble-gro

Reply:
afaik the changes made in #815 make it so only the excluded tokens are excluded while the current code excludes the whole pair if any of it's tokens are on the excluded list. the reason why i made it that way was because i was using it to exclude pool2 pairs, so i wanted to exclude whole pairs if one of the tokens was the gov token